shared_model/backend/protobuf/proposal.hpp

```diff
@@ -21,48 +9,38 @@
 #include "backend/protobuf/transaction.hpp"
 #include "interfaces/iroha_internal/proposal.hpp"
 
-#include "common_objects/trivial_proto.hpp"
+#include "common_objects/noncopyable_proto.hpp"
 
 #include "interfaces/common_objects/types.hpp"
 #include "proposal.pb.h"
 #include "utils/lazy_initializer.hpp"
 
 namespace shared_model {
-
   namespace proto {
-    class Proposal final : public CopyableProto<interface::Proposal,
-                                                iroha::protocol::Proposal,
-                                                Proposal> {
+    class Proposal final : public NonCopyableProto<interface::Proposal,
+                                                   iroha::protocol::Proposal,
+                                                   Proposal> {
      public:
-      template <class ProposalType>
-      explicit Proposal(ProposalType &&proposal)
-          : CopyableProto(std::forward<ProposalType>(proposal)) {}
-
-      Proposal(const Proposal &o) : Proposal(o.proto_) {}
+      using NonCopyableProto::NonCopyableProto;
 
-      Proposal(Proposal &&o) noexcept : Proposal(std::move(o.proto_)) {}
+      Proposal(Proposal &&o) noexcept;
+      Proposal &operator=(Proposal &&o) noexcept;
 
       interface::types::TransactionsCollectionType transactions()
-          const override {
-        return *transactions_;
-      }
+          const override;
 
-      interface::types::TimestampType createdTime() const override {
-        return proto_->created_time();
-      }
+      interface::types::TimestampType createdTime() const override;
 
-      interface::types::HeightType height() const override {
-        return proto_->height();
-      }
+      interface::types::HeightType height() const override;
 
      private:
-      // lazy
       template <typename T>
       using Lazy = detail::LazyInitializer<T>;
 
       const Lazy<std::vector<proto::Transaction>> transactions_{[this] {
-        return std::vector<proto::Transaction>(proto_->transactions().begin(),
-                                               proto_->transactions().end());
+        return std::vector<proto::Transaction>(
```

Comment:
Isn't it possible to move this initialisation to constructor in cpp file as well?

Reply:
Unfortunately no